staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -196,6 +196,22 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
```

Comment:
"MessageExpression is a CEL expression that is evaluated when the rule fails to produce a failure message for the rule." ?

Reply:
Doesn't that make it sound a bit like `MessageExpression` is a fallback when `Message` or the default message don't work for some reason?

Reply:
Yes, that doesn't always parse right. You're fix of putting the "when this rule fails" at the end of the phrase is the correct. fix.